slideCanavs 예외 처리 및 slideCanvas 리팩토링

always-awake opened the pull request:
# slideCanavs 예외 처리 및 slideCanvas 리팩토링
## 해당 이슈 📎
- DropyCanvas 모듈 리팩토링 (#195) 
- 스피커 필기기능 리팩토링(#179)
- [ 리팩토링 ] 캔버스를 그릴 때, canvasHistory 상태 값을 전역으로 관리하지 않도록 리팩토링해야 한다.(#175)

## 변경 사항 🛠
* slideCanvas에서 예외처리 코드를 추가해 주었습니다.
* slideCanvas에서 페이지 이동 시, 초기화 작업 코드를 withInit(HOC)로 분리하였습니다.
    - slideCanvas 내부에는 InitialData를 이용하여, 히스토리 데이터 그리기, 리사이즈 될 때 다시 그리기, 지우기 기능 등의 로직(그리기와 관련된 로직)만 있도록 리팩토링하였습니다.

## 리뷰어 참고 사항 🙋‍♀️
* 더 추가적인 예외사항이 없는지 많은 기능 테스트 부탁드립니다! 🙏🏻

always-awake committed:
[ FIX ] SlideCanvas 코드 전체적으로 리팩토링

Why
- 여러 상황에 대한 예외 처리를 추가해야 한다.

What
- slideCanvas dropyCanavs를 생성하고, useState로 상태관리하도록 수정하였다.
- 지우개 활성화 상태를 channel 리듀서로 관리할 수 있도록 수정하였다.
- initSlideCanvas 로직에 예외처리를 추가해주었다.

always-awake committed:
[ REFACTOR ] SlideCanvas 컴포넌트에서 초기화 하는 로직을 withInit(HOC)로 분리

Why
- SlideCanvas에는 초기화 작업(데이터 베이스에서 히스토리 조회) 로직과 그리기/지우기 로직이 모두 있는데 이를 분리해야할 필요가 있다고 생각함

What
- withInit(HOC)에 page를 이동했을 때, 해당 page의 히스토리를 데이터 베이스에서 조회하는 로직 구현
- withInit은 초기화된 데이터를 SlideCanvas 컴포넌트에 props로 넘겨주도록 구현

happydhKim commented on web/src/components/channel/Slide/SlideViewer/MainSlide/MainSlide.jsx:
```diff
@@ -51,14 +49,12 @@ const MainSlide = (props) => {
       });
     };
-    const throttleResize = throttle(() => handleResize(), 500);
+    const throttleResize = throttle(() => handleResize(), 100);
```
여기서 100은 뭘 의미하는 걸까요?🤔

always-awake replied:
아 throttleResize에 적용할 시간을 의미하는데 매직넘버 같네요 따로 변수화 해 놓겠습니다!

happydhKim reviewed (approved):
와우 주말에 고생하셨어요!!🌝

aereeeee commented on web/src/components/channel/Slide/SlideViewer/MainSlide/SlideCanvas/withInit.jsx:
```diff
@@ -0,0 +1,38 @@
+import { useGetCanvasHistory, useChannelSelector } from '@/hooks';
+import DropyCanvas from '@/utils/DropyCanvas';
+
+const withInit = (WrappedComponent) => (props) => {
```
high order component를 만들어서 비즈니스 로직을 분리하니 훨씬 깔끔하네요! 굳이 아쉬운점이 있다면 이 컴포넌트가 캔버스초기화 로직을 담당하는 만큼 withCanvasInit정도로 이름이 더 구체적인게 좋을것 같습니다 :)

always-awake replied:
넵 더 좋은 이름인 것 같아요!!! 수정하겠습니다 🥰

aereeeee reviewed (approved):
수고하셨습니다! 필기기능 완성도가 높아졌네요👍

always-awake committed:
[ FIX ] 리뷰 사항 반영

Why
- 리뷰어들의 피드백을 받아 이를 수정해야 한다.

What
- 매직넘버 사용을 하지 않고, 상수화 해주었다.
- whitInit 함수를 withCanvasInit명으로 수정하였다.
